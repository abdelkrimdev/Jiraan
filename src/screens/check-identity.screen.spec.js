import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import ConnectedCheckIdentity, { CheckIdentity } from './check-identity.screen';

const mockStore = configureStore([  ]);
const navigation = { navigate: jest.fn() };

describe('check identity screen', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('should render without crashing.', () => {
        const wrapper = shallow(
            <CheckIdentity getAuthenticatedUser={ jest.fn() } navigation={ navigation } />
        );

        expect(wrapper).toBeDefined();
    });

    it('should render correctly.', () => {
        const wrapper = shallow(
            <CheckIdentity getAuthenticatedUser={ jest.fn() } navigation={ navigation } />
        );
        
        expect(wrapper).toMatchSnapshot();
    });

    it('should dispatch the getCurrentUser action when component is mounting.', () => {
        const store = mockStore({
            currentUser: { data: null }
        });

        const container = shallow(<ConnectedCheckIdentity navigation={ navigation } store={ store } />).dive();

        expect(container).toBeDefined();
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should redirect to authentication stack when the user is not logged in.', () => {
        const store = mockStore({
            currentUser: { data: null }
        });

        const container = shallow(<ConnectedCheckIdentity navigation={ navigation } store={ store } />).dive();

        expect(container).toBeDefined();
        expect(navigation.navigate).toHaveBeenCalled();
        expect(navigation.navigate).toHaveBeenCalledWith('Auth');
    });

    it('should redirect to application stack when the user is logged in.', () => {
        const store = mockStore({
            currentUser: { data: {  } }
        });

        const container = shallow(<ConnectedCheckIdentity navigation={ navigation } store={ store } />).dive();

        expect(container).toBeDefined();
        expect(navigation.navigate).toHaveBeenCalled();
        expect(navigation.navigate).toHaveBeenCalledWith('App');
    });
});
