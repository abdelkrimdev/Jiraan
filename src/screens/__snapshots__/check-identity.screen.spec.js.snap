// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check identity screen should dispatch the getCurrentUser action when component is mounting. 1`] = `
Array [
  Object {
    "type": "FIRE_GET_CURRENT_USER_ACTION",
    "user": null,
  },
]
`;

exports[`check identity screen should render correctly. 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
</Component>
`;
